app/code/Magento/CatalogSearch/Test/Unit/Controller/Advanced/ResultTest.php

```diff
@@ -49,4 +52,90 @@ function ($added) use (&$filters) {
+            \Magento\CatalogSearch\Model\Advanced::class,
+            ['addFilters']
+        );
+
+        $catalogSearchAdvanced->expects($this->once())->method('addFilters')->will(
+            $this->throwException(new \Magento\Framework\Exception\LocalizedException(
+                new \Magento\Framework\Phrase("Test Exception")
+            ))
+        );
+
+        $responseMock = $this->createMock(\Magento\Framework\Webapi\Response::class);
+        $requestMock = $this->createPartialMock(
+            \Magento\Framework\App\Request\Http::class,
+            ['getQueryValue']
+        );
+        $requestMock->expects($this->any())->method('getQueryValue')->willReturn(['key' => 'value']);
+
+        $redirectMock = $this->createMock(\Magento\Framework\App\Response\RedirectInterface::class);
+        $redirectMock->expects($this->any())->method('error')->with('urlstring');
+
+        $messageManagerMock = $this->createMock(\Magento\Framework\Message\Manager::class);
+
+        $eventManagerMock = $this->createMock(\Magento\Framework\Event\ManagerInterface::class);
+
+        $contextMock = $this->createMock(\Magento\Framework\App\Action\Context::class);
+        $contextMock->expects($this->any())
+            ->method('getRequest')
+            ->willReturn($requestMock);
+        $contextMock->expects($this->any())
+            ->method('getResponse')
+            ->willReturn($responseMock);
+        $contextMock->expects($this->any())
+            ->method('getRedirect')
+            ->willReturn($redirectMock);
+        $contextMock->expects($this->any())
+            ->method('getMessageManager')
+            ->willReturn($messageManagerMock);
+        $contextMock->expects($this->any())
+            ->method('getEventManager')
+            ->willReturn($eventManagerMock);
+        $contextMock->expects($this->any())
+            ->method('getResultRedirectFactory')
+            ->willReturn($redirectFactoryMock);
+
+        $urlMock = $this->createMock(\Magento\Framework\Url::class);
+        $urlMock->expects($this->once())
+            ->method('addQueryParams')
+            ->willReturnSelf();
+        $urlMock->expects($this->once())
+            ->method('getUrl')
+            ->willReturn("urlstring");
+
+        $urlFactoryMock = $this->createMock(\Magento\Framework\UrlFactory::class);
+        $urlFactoryMock->expects($this->once())
+            ->method('create')
+            ->will($this->returnValue($urlMock));
+
+        $objectManager = new \Magento\Framework\TestFramework\Unit\Helper\ObjectManager($this);
+
+        /** @var \Magento\CatalogSearch\Controller\Advanced\Result $instance */
+        $instance = $objectManager->getObject(
+            \Magento\CatalogSearch\Controller\Advanced\Result::class,
+            ['context' => $contextMock,
+            'catalogSearchAdvanced' => $catalogSearchAdvanced,
+            'urlFactory' => $urlFactoryMock
+            ]
+        );
+        $instance->execute();
```

Comment:
I think it will be good idea to check if method returned $responseMock

Reply:
@ihor-sviziev - Done, pushed, and pending automated testing.